encryption/propagator.go

```diff
@@ -0,0 +1,83 @@
+package encryption
+
+import (
+	"context"
+
+	"go.temporal.io/sdk/converter"
+	"go.temporal.io/sdk/workflow"
+)
+
+type (
+	// contextKey is an unexported type used as key for items stored in the
+	// Context object
+	contextKey struct{}
+
+	// propagator implements the custom context propagator
+	propagator struct{}
+
+	// CryptContext is a struct holding values
+	CryptContext struct {
+		Key string `json:"key"`
+	}
+)
+
+// PropagateKey is the key used to store the value in the Context object
+var PropagateKey = contextKey{}
```

Comment:
We do not use this field anywhere in other packages, so what about `contextKey{}` for each usages?

As a side note, the final dots are missing for godocs here.